### PR TITLE
Security and Documentation

### PR DESCRIPTION
**Pull request by friedger**
This PR
* secures pool operator's functions with (check-contract-caller)
* adds more comments
* regroups functions
* improves function names

### CODE REVIEW

File: contracts/pox-delegation.clar [hunk 1]
```diff
@@ -1,16 +1,29 @@
-(define-trait pool-trait-ext ((delegate-stx (uint principal (optional uint)
-              (optional (tuple (hashbytes (buff 32)) (version (buff 1))))
-              (tuple (hashbytes (buff 32)) (version (buff 1)))
-              uint) (response bool int))))
+;; @contract PoX-2 wrapper contract for stacking pools
+;; @version 1
 
+;; User calls delegate-stx at first and provide a btc address to receive rewards.
```
Comment (2snEM6) [inline contracts/pox-delegation.clar:4]:
```suggestion
;; User calls delegate-stx at first and provides a btc address to receive rewards.
```

File: contracts/pox-delegation.clar [hunk 2]
```diff
@@ -34,7 +42,7 @@
 (define-private (pox-get-stacker-info (user principal))
    (contract-call? 'SP000000000000000000002Q6VF78.pox-2 get-stacker-info user))
 
-
+;; Get currently delegated amount
```
Comment (2snEM6) [inline contracts/pox-delegation.clar]: `for a given principal`?

**Review by 2snEM6 — COMMENTED** [review]
LGTM

**Comment by friedger** [comment]
comments addressed

**Pull request merged**